source/openqasm/openqasm3/printer.py

```diff
@@ -243,6 +255,12 @@ def visit_Program(self, node: ast.Program, context: PrinterState) -> None:
         for statement in node.statements:
             self.visit(statement, context)
 
+    @_maybe_annotated
+    def visit_CompoundStatement(self, node: ast.CompoundStatement, context: PrinterState) -> None:
+        self._start_line(context)
+        self._visit_statement_list(node.statements, context)
```

Comment:
```suggestion
        self._visit_statement_list(node.statements, context, prefix="")
```

We shouldn't indent this beyond the current printer position.